lib/smart_answer_flows/calculate-your-holiday-entitlement.rb

```diff
@@ -154,13 +169,16 @@ def define
 
       # Q10 - Q15 - Q18
       value_question :how_many_hours_per_week?, parse: Float do
-        calculate :hours_per_week do |response|
-          hours_per_week = response
-          raise InvalidResponse, :error_over_168_hours_worked if hours_per_week > 168
+        on_response do |response|
+          calculator.hours_per_week = response
+        end
 
-          raise InvalidResponse, :error_no_hours_worked if hours_per_week <= 0
+        validate :error_no_hours_worked do
+          calculator.hours_per_week.positive?
+        end
 
-          hours_per_week
+        validate :error_over_168_hours_worked do
+          calculator.hours_per_week.between?(1, 168)
```

Comment:
Again, `between?` is great where the range is being caught within one validation, but when it is split into two (lower and higher limits), the test for the higher limit should just test that, and not use `between?`.

Reply:
Updated as per suggestion - thanks 👍